lib/http-proxy/common.js

```diff
@@ -41,7 +41,24 @@ common.setupOutgoing = function(outgoing, options, req, forward) {
   );
 
   outgoing.method = req.method;
-  outgoing.headers = extend({}, req.headers);
+
+  var connectionHeaderName = 'connection';
+  var hostHeaderName = 'host';
+  if (options.preserveHeaderKeyCase && req.rawHeaders != undefined) {
+    var rawHeaders = {};
+    for (var i = 0; i < req.rawHeaders.length - 1; i += 2) {
+      var key = req.rawHeaders[i];
+      rawHeaders[key] = req.rawHeaders[i + 1];
```

Comment:
Learning from #1104, I prefer to use `req.headers[key]` instead.
I don't check the case of incoming, though.